src/commands/data/import/bulk.ts

```diff
@@ -0,0 +1,246 @@
+    }),
+    sobject: Flags.string({
+      summary: messages.getMessage('flags.sobject.summary'),
+      char: 's',
+      required: true,
+    }),
+    'api-version': Flags.orgApiVersion(),
+    wait: Flags.duration({
+      summary: messages.getMessage('flags.wait.summary'),
+      char: 'w',
+      unit: 'minutes',
+      exclusive: ['async'],
+    }),
+    'target-org': Flags.requiredOrg(),
+    'line-ending': Flags.option({
+      summary: messages.getMessage('flags.line-ending.summary'),
+      dependsOn: ['file'],
+      options: ['CRLF', 'LF'] as const,
+    })(),
+  };
+
+  public async run(): Promise<DataImportBulkResult> {
+    const { flags } = await this.parse(DataImportBulk);
+
+    const conn = flags['target-org'].getConnection(flags['api-version']);
+
+    const timeout = flags.async ? Duration.minutes(0) : flags.wait ?? Duration.minutes(0);
+    const async = timeout.milliseconds === 0;
+
+    const baseUrl = flags['target-org'].getField<string>(Org.Fields.INSTANCE_URL).toString();
+
+    const ms = new MultiStageOutput<JobInfoV2>({
+      jsonEnabled: flags.json ?? false,
+      stages: async ? ['Creating ingest job'] : ['Creating ingest job', 'Processing the job'],
+      title: async ? 'Importing data (async)' : 'Importing data',
+      stageSpecificBlock: [
+        {
+          stage: 'Processing the job',
+          label: 'Processed records',
+          type: 'dynamic-key-value',
+          get: (data): string | undefined => {
+            if (data?.numberRecordsProcessed) {
+              return data.numberRecordsProcessed.toString();
+            }
+          },
+        },
+        {
+          stage: 'Processing the job',
+          label: 'Successful records',
+          type: 'dynamic-key-value',
+          get: (data): string | undefined => {
+            const numberRecordsFailed = data?.numberRecordsFailed ?? 0;
+
+            if (data?.numberRecordsProcessed) {
+              return (data.numberRecordsProcessed - numberRecordsFailed).toString();
+            }
+          },
+        },
+        {
+          stage: 'Processing the job',
+          label: 'Failed records',
+          type: 'dynamic-key-value',
+          get: (data): string | undefined => {
+            if (data?.numberRecordsFailed) {
+              return data.numberRecordsFailed.toString();
+            }
+          },
+        },
+      ],
+      postStagesBlock: [
+        {
+          label: 'Status',
+          type: 'dynamic-key-value',
+          bold: true,
+          get: (data) => data?.state,
+        },
+        {
+          label: 'Job Id',
+          type: 'dynamic-key-value',
+          bold: true,
+          get: (data) =>
+            data?.id &&
+            terminalLink(
+              data.id,
+              `${baseUrl}/lightning/setup/AsyncApiJobStatus/page?address=${encodeURIComponent(`/${data.id}`)}`
+            ),
+        },
+      ],
+    });
+
+    if (async) {
+      ms.goto('Creating ingest job');
```

Comment:
Do you need to use `.skipTo` or `.goto`?

`.goto` is going to mark any skipped stages as `complete` whereas `.skipTo` will mark any skipped stages as `skipped`

Reply:
in async there's only one stage:
https://github.com/salesforcecli/plugin-data/blob/be3e6686163df2f823dfe240f8f9402c9e622ef1/src/commands/data/import/bulk.ts#L78

But I think you are right, we can leave all stages and skip them instead of not rendering them (`import resume` does this) 👍🏼